### PR TITLE
scripts: some cleanup + reboot sequence generation moved to own file

### PR DESCRIPTION
**Pull request by cruwaller**

### CODE REVIEW

File: src/python/BFinitPassthrough.py [hunk 1]
```diff
@@ -105,38 +105,45 @@ def bf_passthrough_init(port, requestedBaudrate, half_duplex=False, reset_to_bl=
     dbg_print("Enabling serial passthrough...")
     dbg_print("  CMD: '%s'" % cmd)
     rl.write(cmd + '\n')
-
-    dbg_print("======== PASSTHROUGH DONE ========")
-
-    if(reset_to_bl):
```
Comment (cruwaller) [inline src/python/BFinitPassthrough.py:111]: this is moved into main function to enable passthrough flashing using USB FTDI adapter.

File: src/python/UARTupload.py [hunk 2]
```diff
@@ -215,9 +219,11 @@ def on_upload(source, target, env):
         for flag in flags:
             if "GHST=" in flag:
                 ghst = eval(flag.split("=")[1])
+            elif "BL_KEY=" in flag:
```
Comment (cruwaller) [inline src/python/UARTupload.py:222]: BL_KEY will be used to validate correct reboot command per env.

File: src/python/stlink.py [hunk 3]
```diff
@@ -43,13 +43,6 @@ def get_commands(env, firmware):
         if bootloader is not None:
             BL_CMD = [TOOL, "write", bootloader, hex(flash_start)]
         APP_CMD = [TOOL, "--reset", "write", firmware, hex(app_start)]
-    elif "darwin" in platform_name:
```
Comment (cruwaller) [inline src/python/stlink.py:46]: useless branch. Darwin is handled above.

**Comment by CapnBry** [comment]
While you're making these changes to the python, can you change this line here [BFinitPassthrough.py#L107](https://github.com/ExpressLRS/ExpressLRS/blob/932095efd65c9cd30f19d555c29073f7e05aba5b/src/python/BFinitPassthrough.py#L107) to `rl.write(cmd + '\r\n')`? It is the only place we don't send `\r\n` for a command and it caught me by surprise when I added passthrough flashing to CRServoF.

I'd also make the reset_to_bl a new function instead of tacking it on at the end.

Finally, why is the key fixed as "ESP82" when different MCUs can use passthrough flashing? How is the key going to be used in RX?

**Comment by cruwaller** [comment]
1) Why do we want to add `\r\n`? It does not send strings so not needed. Caller should take care of the line ends if needed, imo :)
2) new func makes sense. I will move it.
3) `__main__ ` is used only for ESP passthrough so it defaults to that. I will add a argument for type :)

**Comment by CapnBry** [comment]
> 1. Why do we want to add `\r\n`? It does not send strings so not needed. Caller should take care of the line ends if needed, imo :)

Look at the line I linked. It adds + '\n'. Every other command sent to betaflight CLI ends with '\r\n'. The caller can't append its own '\r', this is internal to the function. It is just a mistake, someone left it off when they coded this.

File: src/python/esp_compress.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 import shutil
 import os, glob
 
-FIRMWARE_PACKING_ENABLED = False
+FIRMWARE_PACKING_ENABLED = True
```
Comment (cruwaller) [inline src/python/esp_compress.py:5]: Compress will generate another file with `bin.gz` extension so this can be enabled now.

**Comment by cruwaller** [comment]
> > 1. Why do we want to add `\r\n`? It does not send strings so not needed. Caller should take care of the line ends if needed, imo :)
> 
> Look at the line I linked. It is a string command to betaflight and adds + '\n'. Every other command sent to betaflight CLI ends with '\r\n'. The caller can't append its own '\r', this is internal to the function. It is just a mistake, someone left it off when they coded this.

Oh true! My bad... Will fix those as well 👍

**Comment by CapnBry** [comment]
Yep that all checks out and I see the type arg now that you've mentioned it. Looks good to me.

**Review by CapnBry — APPROVED** [review]

**Pull request merged**